WIP: readme restructure (#32)
openshift/csi-driver-projected-resource: README.md

> # OpenShift Projected Resource "CSI DRIVER"

Reviewer: Remove quotation marks. If the secret store CSI driver is a CSI driver, so is this.
```suggestion
# OpenShift Projected Resource CSI Driver
```

  
## Features

- Supports only a subset of the Kubernetes CSIVolumeSource API.  See [CSI Volume Specifics](docs/csi.md) for details
- Initial pod requests for `Share` CSI volumes are denied without both a valid `Share` reference and
  permissions to access that `Share`
- Changes to the `Share`'s backing resource (kind, namespace, name) get reflected in data stored in the user pod's CSI volume
- Subsequent removal of permissions for a `Share` results in removal of the associated data stored in the user pod's CSI volume
- Re-granting of permission for a `Share` (after having the permissions initially, then removed) results in the associated
  data getting stored in the user pod's CSI volume
- Removal of the `Share` used to provision `Share` csi volume for a pod result in the associated data getting removed  
- Re-creation of a removed `Share` for a previously provisioned `Share` CSI volume results in the associated data
  reappearing in the user pod's CSI volume
- Supports recycling of the csi driver so that previously provisioned CSI volumes are still managed; in other words,
  the driver's interan state is persisted

> - Multiple `Share`s within a pod are allowed

Reviewer: format nit
```suggestion
- Multiple `Shares` within a pod are allowed
```

> - When multiple `Share`s are mounted in a pod, one `Share` can be mounted as a subdirectory of another `Share`

Reviewer: format nit
```suggestion
- When multiple `Shares` are mounted in a pod, one `Share` can be mounted as a subdirectory of another `Share`
```

  

## Getting Started

Check out the [current installation options](docs/install.md) to get the driver up and going.  You'll need to have
sufficient privileges to create namespaces and ServiceAccounts, and then create `ClusterRoles`, `ClusterRoleBindings`, `DaemonSets` with the privileged bit set,
and the creation of `CSIDrivers`.

Then, check out our [entry level example](docs/simple-example.md).  You'll need to have sufficient privileges to create
namespaces, `ClusterRoles` and `ClusterRoleBindings`, instances of our new `Share` CRD, and pods.

> The new `Share` CRD is [here](deploy/deploy/0000_10_projectedresource.crd.yaml)

Reviewer: Fixing the link, clarifying what the link will show:
```suggestion
The full definition of the `Share` custom resource is [here](deploy/0000_10_projectedresource.crd.yaml)
```
